app/javascript/packs/application.js

```diff
@@ -6,6 +6,14 @@ import { Provider } from "react-redux";
 import App from "./application/App";
 import store from "./application/redux";
 
+// Requirements for foundation
+import $ from "jquery";
+import "foundation-sites";
+import "./application/styles/index.scss";
+
+// Kick off foundation
+$(document).foundation();
```

Comment:
I think you can remove the jQuery stuff here, because you've added react-foundation. If `react-foundation` depends on jQuery, then you simply shouldn't use it, because it isn't providing any value.